### PR TITLE
Run device discovery in a loop

### PR DESCRIPTION
**Pull request by SchSeba**
Create a loop for the device discovery.

This PR also changes the Dockerfiles to use the vendor folder on the build and not fetch the modules.

Related to the follow error:
```
PTP capable NICs: [eno1 eno2]
E0418 22:40:28.559220    4744 main.go:84] failed to get NodePtpDevice CR for node cnfd1-worker-0.fci1.kni.lab.eng.bos.redhat.com: Get https://172.30.0.1:443/apis/ptp.openshift.io/v1/namespac
es/openshift-ptp/nodeptpdevices/cnfd1-worker-0.fci1.kni.lab.eng.bos.redhat.com: dial tcp 172.30.0.1:443: i/o timeout
I0418 22:40:28.587016    4744 ptpdev.go:15] PTP capable NICs: [eno1 eno2]
E0418 22:40:28.587075    4744 main.go:93] failed to update Node PTP device CR: resource name may not be empty
I0418 22:40:58.589483    4744 main.go:122] ticker pull
```

### CODE REVIEW

File: pkg/daemon/ptpdev.go [hunk 1]
```diff
@@ -28,3 +33,38 @@ func GetDevStatusUpdate(nodePTPDev *ptpv1.NodePtpDevice) (*ptpv1.NodePtpDevice,
...
+
+	// Assume NodePtpDevice CR for this particular node
+	// is already created manually or by PTP-Operator.
+	ptpDev, err := ptpClient.PtpV1().NodePtpDevices(PtpNamespace).Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		glog.Errorf("failed to get NodePtpDevice CR for node %s: %v", nodeName, err)
+	}
+
+	// Render status of NodePtpDevice CR by inspecting PTP capability of node network devices
+	ptpDev, _ = GetDevStatusUpdate(ptpDev)
```
Comment (fedepaol) [inline pkg/daemon/ptpdev.go:54]: Can you check & log the error here?

File: pkg/daemon/ptpdev.go [hunk 2]
```diff
@@ -28,3 +33,38 @@ func GetDevStatusUpdate(nodePTPDev *ptpv1.NodePtpDevice) (*ptpv1.NodePtpDevice,
...
+
+	// Update NodePtpDevice CR
+	_, err = ptpClient.PtpV1().NodePtpDevices(PtpNamespace).UpdateStatus(ptpDev)
```
Comment (fedepaol) [inline pkg/daemon/ptpdev.go:57]: Does it make sense to update the status only if it's different from the current one?

File: pkg/daemon/ptpdev.go (hunk 1 above)
Comment (SchSeba) [reply pkg/daemon/ptpdev.go:54]: done

File: pkg/daemon/ptpdev.go (hunk 2 above)
Comment (SchSeba) [reply pkg/daemon/ptpdev.go:60]: there is room for improvements here for sure but I will like to do it on a different PR if that is fine by you

**Pull request merged**